conf/modules.config

```
@@ -951,40 +955,79 @@ process {
     // LIBRARY MERGE
     //
 
-    withName: SAMTOOLS_MERGE_LIBRARIES {
+    withName: ".*MERGE_LIBRARIES:SAMTOOLS_MERGE_LIBRARIES" {
+        tag = { "${meta.reference}|${meta.sample_id}" }
+        ext.prefix = { "${meta.sample_id}_${meta.reference}_unsorted" }
+        publishDir = [
+            enabled: false
+        ]
+    }
+
+    withName: ".*MERGE_LIBRARIES:SAMTOOLS_SORT_MERGED_LIBRARIES" {
+        tag = { "${meta.reference}|${meta.sample_id}" }
+        ext.prefix = { "${meta.sample_id}_${meta.reference}" }
+        publishDir = [
+            path: { "${params.outdir}/final_bams/raw/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.bam'
+        ]
+    }
+
+    withName: ".*MERGE_LIBRARIES:SAMTOOLS_INDEX_MERGED_LIBRARIES" {
+        tag = { "${meta.reference}|${meta.sample_id}" }
+        ext.args = { params.fasta_largeref ? "-c" : "" }
+        ext.prefix = { "${meta.sample_id}_${meta.reference}" }
+        publishDir = [
+            path: { "${params.outdir}/final_bams/raw/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.{bai,csi}'
+        ]
+    }
+
+    withName: ".*MERGE_LIBRARIES:SAMTOOLS_FLAGSTAT_MERGED_LIBRARIES" {
+        tag = { "${meta.reference}|${meta.sample_id}" }
+        ext.prefix = { "${meta.sample_id}_${meta.reference}" }
+        publishDir = [
+            path: { "${params.outdir}/final_bams/raw/" },
+            mode: params.publish_dir_mode,
+            pattern: '*.flagstat'
+        ]
+    }
+
+        withName: ".*MERGE_LIBRARIES:SAMTOOLS_MERGE_LIBRARIES" {
         tag = { "${meta.reference}|${meta.sample_id}" }
         ext.prefix = { "${meta.sample_id}_${meta.reference}_unsorted" }
         publishDir = [
             enabled: false
         ]
     }
 
-    withName: SAMTOOLS_SORT_MERGED_LIBRARIES {
+    withName: ".*MERGE_LIBRARIES_GENOTYPING:SAMTOOLS_SORT_MERGED_LIBRARIES" {
         tag = { "${meta.reference}|${meta.sample_id}" }
         ext.prefix = { "${meta.sample_id}_${meta.reference}" }
         publishDir = [
-            path: { "${params.outdir}/final_bams/" },
+            path: { "${params.outdir}/final_bams/for_genotyping/" },
```

```suggestion
            path: { "${params.outdir}/final_bams/${params.genotyping_source}/" },
```
I think this should reflect the source of the bam file, as that is cleaner.